src/controllers/v1/attachment/index.ts

```diff
@@ -223,21 +223,26 @@ export class AttachmentController extends Controller {
     @Path() idOrHash: AttachmentIdOrHash
   ): Promise<unknown | Readable> {
     this.log.debug(`attempting to retrieve ${idOrHash} attachment`)
-
+    let blobBuffer: Buffer<ArrayBuffer> | null = null
     const attachment = await this.findAttachmentRecord(idOrHash, req.user)
-
-    const { blob, filename: ipfsFilename } = await this.ipfs.getFile(attachment.integrity_hash)
-    const blobBuffer = Buffer.from(await blob.arrayBuffer())
-
+    const self = await this.identity.getMemberBySelf()
+    if (attachment.owner !== self.address) {
+      blobBuffer = await this.getAttachmentFromPeer(attachment)
+    }
     let { filename, size } = attachment
-    if (size === null || filename === null) {
-      try {
-        await this.db.update('attachment', { id: attachment.id }, { filename: ipfsFilename, size: blob.size })
-        filename = ipfsFilename
-        size = blob.size
-      } catch (err) {
-        const message = err instanceof Error ? err.message : 'unknown'
-        this.log.warn('Error updating attachment size: %s', message)
+    if (blobBuffer === null) {
+      const { blob, filename: ipfsFilename } = await this.ipfs.getFile(attachment.integrity_hash)
+      blobBuffer = Buffer.from(await blob.arrayBuffer())
+
+      if (size === null || filename === null) {
+        try {
+          await this.db.update('attachment', { id: attachment.id }, { filename: ipfsFilename, size: blob.size })
+          filename = ipfsFilename
+          size = blob.size
+        } catch (err) {
+          const message = err instanceof Error ? err.message : 'unknown'
+          this.log.warn('Error updating attachment size: %s', message)
+        }
```

Comment:
I'd refactor this our out to a new function which returns the `blockBuffer` and `filename` based on an attachment record. That way you can early return out and you can avoid `let` and reduce the amount of indentation. 

For the external case the filename will be provided in the response so you should be able to use that rather than "external"